pkg/rancher-desktop/config/__tests__/settings.spec.ts

```diff
@@ -72,10 +249,255 @@ describe('updateFromCommandLine', () => {
+          settings.load();
+          verifyAllFieldsAreLocked(settings.getLockedSettings());
+        });
+      });
+
+      describe('when there is only a system profile with unlocked imageList', () => {
+        beforeEach(() => {
+          mock = jest.spyOn(fs, 'readFileSync')
+            .mockImplementation(createMocker(1, 0));
+        });
+        afterEach(() => {
+          mock.mockRestore();
+        });
+        test('all fields are unlocked', () => {
+          settings.load();
+          verifyAllFieldsAreUnlocked(settings.getLockedSettings());
+        });
+      });
+
+      describe('when both profiles exit, both with unlocked imageList', () => {
+        beforeEach(() => {
+          mock = jest.spyOn(fs, 'readFileSync')
+            .mockImplementation(createMocker(1, 1));
+        });
+        afterEach(() => {
+          mock.mockRestore();
+        });
+        test('all fields are locked', () => {
+          settings.load();
+          verifyAllFieldsAreUnlocked(settings.getLockedSettings());
+        });
+      });
+
+      describe('when the system profile is unlocked and the user profile is locked', () => {
+        beforeEach(() => {
+          mock = jest.spyOn(fs, 'readFileSync')
+            .mockImplementation(createMocker(1, 2));
+        });
+        afterEach(() => {
+          mock.mockRestore();
+        });
+        test('all fields are locked', () => {
+          settings.load();
+          verifyAllFieldsAreUnlocked(settings.getLockedSettings());
+        });
+
+        describe('when there is only a system profile with locked imageList', () => {
+          beforeEach(() => {
+            mock = jest.spyOn(fs, 'readFileSync')
+              .mockImplementation(createMocker(2, 0));
+          });
+          afterEach(() => {
+            mock.mockRestore();
+          });
+          test('all fields are locked', () => {
+            settings.load();
+            verifyAllFieldsAreLocked(settings.getLockedSettings());
+          });
+        });
+
+        describe('when both profiles exit, system locked, user unlocked', () => {
+          beforeEach(() => {
+            mock = jest.spyOn(fs, 'readFileSync')
+              .mockImplementation(createMocker(2, 1));
+          });
+          afterEach(() => {
+            mock.mockRestore();
+          });
+          test('all fields are locked', () => {
+            settings.load();
+            verifyAllFieldsAreLocked(settings.getLockedSettings());
+          });
+        });
+
+        describe('when both profiles exist and are locked', () => {
+          beforeEach(() => {
+            mock = jest.spyOn(fs, 'readFileSync')
+              .mockImplementation(createMocker(2, 2));
+          });
+          afterEach(() => {
+            mock.mockRestore();
+          });
+          test('all fields are locked', () => {
+            settings.load();
+            verifyAllFieldsAreLocked(settings.getLockedSettings());
```

Comment:
Should we be checking that the system-level settings win?

Reply:
This is tested here:
```
when the system profile is unlocked and the user profile is locked
verifyAllFieldsAreUnlocked...
```
and here:
```
when both profiles exit, system locked, user unlocked
verifyAllFieldsAreLocked
```